src/catch.rs

```diff
@@ -0,0 +1,92 @@
+use std::marker::PhantomData;
+
+use jni::{
+    objects::{AutoLocal, JObject},
+    JNIEnv,
+};
+
+use crate::{cast::Upcast, error::Error, java::lang::Throwable, JvmOp, Local};
+
+/// Plumbing utility to check the exception state of the JVM thread convert it into a [`crate::Result`].
+// XX: many of the jni crate checked methods do this automatically. We only need this if/when we invoke
+// unchecked methods.
+pub fn try_catch<'jvm>(env: &mut JNIEnv<'jvm>) -> crate::Result<'jvm, ()> {
```

Comment:
I don't see any calls to this function?

Reply:
Yep, I had written it assuming the `*_unchecked` version of JNI functions wouldn't check for the exception flag and we'd use this in the next PR that moves to `*_unchecked`, but that was wrong. It looks like it's checked everywhere.

Will delete this!

Reply:
Coming back, I think we actually need this after our `jni` crate calls. A user that inspects the error case of `.execute()`  inside of `Jvm::with` won't get the materialized thrown object. We can't do the materialization by default in the `From` impl either, because it needs a `&mut Jvm` to get the reference.